src/ParallelAlgorithms/Events/ObserveTimeStep.hpp

```diff
@@ -0,0 +1,182 @@
+struct TimeStep;
+}  // namespace Tags
+/// \endcond
+
+namespace Events {
+template <typename Metavariables, typename EventRegistrars>
+class ObserveTimeStep;
+
+namespace Registrars {
+template <typename Metavariables>
+using ObserveTimeStep =
+    ::Registration::Registrar<Events::ObserveTimeStep, Metavariables>;
+}  // namespace Registrars
+
+/*!
+ * \brief %Observe the size of the time steps.
+ *
+ * Writes reduction quantities:
+ * - `%Time`
+ * - `NumberOfPoints`
+ * - `%Slab size`
+ * - `Minimum time step`
+ * - `Maximum time step`
+ * - `Effective time step`
+ *
+ * The effective time step is the step size of a global-time-stepping
+ * method that would perform a similar amount of work.
+ *
+ * All values are reported as positive numbers, even for backwards
+ * evolutions.
+ */
+template <typename Metavariables,
+          typename EventRegistrars =
+              tmpl::list<Registrars::ObserveTimeStep<Metavariables>>>
+class ObserveTimeStep : public Event<EventRegistrars> {
+ private:
+  using ReductionData = Parallel::ReductionData<
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<size_t, funcl::Plus<>>,
+      Parallel::ReductionDatum<double, funcl::AssertEqual<>>,
+      Parallel::ReductionDatum<double, funcl::Min<>>,
+      Parallel::ReductionDatum<double, funcl::Max<>>,
+      Parallel::ReductionDatum<
+          double, funcl::Plus<>,
+          funcl::Divides<funcl::Literal<1, double>, funcl::Divides<>>,
+          std::index_sequence<1>>>;
+
+ public:
+  /// The name of the subfile inside the HDF5 file
+  struct SubfileName {
+    using type = std::string;
+    static constexpr Options::String help = {
+        "The name of the subfile inside the HDF5 file without an extension and "
+        "without a preceding '/'."};
+  };
+
+  /// \cond
+  explicit ObserveTimeStep(CkMigrateMessage* /*unused*/) noexcept {}
+  using PUP::able::register_constructor;
+  WRAPPED_PUPable_decl_template(ObserveTimeStep);  // NOLINT
+  /// \endcond
+
+  using options = tmpl::list<SubfileName>;
+  static constexpr Options::String help =
+      "Observe the size of the time steps.\n"
+      "\n"
+      "Writes reduction quantities:\n"
+      "- Time\n"
+      "- NumberOfPoints\n"
+      "- Slab size\n"
+      "- Minimum time step\n"
+      "- Maximum time step\n"
+      "- Effective time step\n"
+      "\n"
+      "The effective time step is the step size of a global-time-stepping\n"
+      "method that would perform a similar amount of work.\n"
+      "\n"
+      "All values are reported as positive numbers, even for backwards\n"
+      "evolutions.";
+
+  ObserveTimeStep() = default;
+  explicit ObserveTimeStep(const std::string& subfile_name) noexcept;
+
+  using observed_reduction_data_tags =
+      observers::make_reduction_data_tags<tmpl::list<ReductionData>>;
+
+  using argument_tags =
+      tmpl::list<Tags::Time, Tags::TimeStep,
+                 typename Metavariables::system::variables_tag>;
```

Comment:
Could you add a C++ comment as to why you are using the Variables instead of the Mesh to get the number of grid points?